packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,331 @@
+
+For icons, if multiple values map the to same icon, it is possible that sorting will result in the instances of a certain icon not being all together in one span of rows. The user may think they should be all together, but this is a corner case that we can't/shouldn't do anything about.
+
+### Sizing
+
+`nimble-table-column-icon` will support only a fixed width. We will introduce a new mixin for fixed-width support that exposes a `pixel-width` property. The default value will be the minimum supported by the table, which is still significantly larger than the width of an icon.
+
+`nimble-table-column-mapping` will support fixed or fractional widths. If `pixel-width` is set, the column will have a fixed width, otherwise it defaults to a fractional width of 1. The client may configure `fractional-width` and/or `min-pixel-width`.
+
+### Angular integration
+
+Angular directives will be created for the column components and the mapping components. No component has form association, so a `ControlValueAccessor` will not be created.
+
+### Blazor integration
+
+Blazor wrappers will be created for the components. Columns will be generic in the type of the key, and will cascade that type parameter to contained mapping elements (see [`CascadingTypeParameter`](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/generic-type-support?view=aspnetcore-7.0#cascaded-generic-type-support)):
+
+```HTML
+<NimbleTableColumnMapping TKey=int Field="NumberData">
+    <NimbleMappingText Key="1" Label="foo"></NimbleMappingText>
+    <NimbleMappingText Key="2" Label="bar"></NimbleMappingText>
+</NimbleTableColumnMapping>
+```
+
+### Visual Appearance
+
+The cell view (and group header view) will be responsible for styling the templates returned by the mappings. This will include alignment and spacing (`--ni-nimble-small-padding`). Unfortunately, because the mappings cannot provide CSS to go with the templates they return, some implementation knowledge will leak from the mappings to the cell/group header views. For example, we must set `flex-shrink: 0` on all elements so that icons do not get smaller in a group header when the adjacent label takes up all the space.
```

Comment:
I don't think I understand the concerns being expressed here but it also feels like implementation details that could be discussed in review. If you feel like it should have additional discussion to capture in the HLD to avoid PR churn we can discuss it.

Reply:
I confused Molly with this same concern. Maybe we don't need to discuss this implementation detail until the actual implementation PR.

Reply:
Can we remove the last part of this paragraph then? I think the first part makes sense, but if we all agree that the leaking of implementation details is an implementation detail, then I'd vote for taking it out of this document.

Reply:
Sure. Removed.